packages/block-library/src/image/index.js

```diff
@@ -65,10 +59,37 @@ const blockAttributes = {
 	height: {
 		type: 'number',
 	},
+	fileWidth: {
+		type: 'number',
+	},
+	fileHeight: {
+		type: 'number',
+	},
+	userWidth: {
+		type: 'number',
+	},
+	userHeight: {
+		type: 'number',
+	},
+	editWidth: {
+		type: 'number',
+	},
 	linkDestination: {
 		type: 'string',
 		default: 'none',
 	},
+	srcSet: {
```

Comment:
We should be able to determine the `srcset` dynamically based on the current image's `id` and size (i.e. thumbnail, default, etc. – which might be a useful attribute). The `sizes` can be represented as a function of the image's width attribute.

Are there other reasons we need to store `srcset` and `sizes` as an attribute on this block and why we would need to save those attributes to the image markup? I would strongly like to avoid doing so.

Reply:
Storing them is for the editor's sake, for faster loading of the content, etc. Both of these are always replaced when rendering the block on the front-end. Retrieving "new" srcset every time a post is opened for editing would slow down loading/opening of the editor and has no benefits.

Reply:
Great. In that case, I think these should be stored in the block delimiter instead of in the saved markup. Even so, I'm still seeing double downloads happening in the editor, so something else is going on here.

It's also worth noting that since images in the editor are rendered as React components and not as part of the HTML that is loaded in the webpage, it opens up different implementation options for doing responsive images as `srcset` and `sizes` was specifically invented to fix the problem of browser pre-loaders fetching images before JS was executed on the page. Not saving these attributes to the markup give us more flexibility, IMO.

Reply:
"for the editor's sake" is not something I think we should want as a reason for an attribute to exist. As I see it, the attributes should contain the minimal normalized data to be able to reconstruct the image in any context. `fileWidth`, `fileHeight`, `srcSet`, `sizes` are redundant if the `id` is already known.